modules/generator/instance.go

```diff
@@ -124,13 +124,47 @@ func (i *instance) watchOverrides() {
 	}
 }
 
+// Look at the processors defined and see if any are actually span-metrics subprocessors
+// If they are, set the appropriate flags in the spanmetrics struct
+func (i *instance) updateSubprocessors(desiredProcessors map[string]struct{}, desiredCfg ProcessorConfig) (map[string]struct{}, ProcessorConfig) {
+	_, allOk := desiredProcessors["span-metrics"]
+	_, countOk := desiredProcessors["span-metrics-count"]
+	_, latencyOk := desiredProcessors["span-metrics-latency"]
+
+	if allOk {
```

Comment:
What's the difference between `allOk` and `countOk && latencyOk`? Are they not equivalent?

Reply:
@mapno In the case of the overrides, you could have the values "span-metrics", "span-metrics-latency", or "span-metrics-count". In this conditional block, ``allOk`` is the case where the user indicated "span-metrics". ``countOk`` and ``latencyOk`` is the case where the user indicated "span-metrics-latency" and "span-metrics-count" separately.

Reply:
I changed the logic here a little bit. Since we know we want to collapse "span-metrics-latency" and "-count" into "span-metrics", we don't need the ``allOk``, and we don't need individual delete statements since we can delete them agnostic to the different conditions